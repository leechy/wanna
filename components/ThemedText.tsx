import { Text, type TextProps, StyleSheet } from 'react-native';

import { useThemeColor } from '@/hooks/useThemeColor';

export type ThemedTextProps = TextProps & {
  lightColor?: string;
  darkColor?: string;
  type?: 'default' | 'title' | 'completed-title' | 'defaultSemiBold' | 'subtitle' | 'link';
};

export function ThemedText({ style, lightColor, darkColor, type = 'default', ...rest }: ThemedTextProps) {
  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');
  const primaryColor = useThemeColor({}, 'primary');
  const touchableColor = useThemeColor({}, 'touchable');

  return (
    <Text
      style={[
        { color: type === 'title' ? primaryColor : color },
        type === 'default' ? styles.default : undefined,
        type === 'title' ? styles.title : undefined,
        type === 'completed-title' ? [styles.title, { color: touchableColor }] : undefined,
        type === 'defaultSemiBold' ? styles.defaultSemiBold : undefined,
        type === 'subtitle' ? styles.subtitle : undefined,
        type === 'link' ? styles.link : undefined,
        style,
      ]}
      numberOfLines={type === 'title' || type === 'completed-title' ? 1 : 0}
      adjustsFontSizeToFit={type === 'title' || type === 'completed-title'}
      {...rest}
    />
  );
}

const styles = StyleSheet.create({
  default: {
    fontFamily: 'Montserrat',
    fontSize: 17,
    lineHeight: 24,
    fontWeight: '500',
    textAlign: 'left',
    letterSpacing: -0.2,
  },
  defaultSemiBold: {
    fontFamily: 'Montserrat',
    fontSize: 18,
    lineHeight: 24,
    fontWeight: '700',
    textAlign: 'left',
  },
  title: {
    fontFamily: 'GreatVibes-Regular',
    fontSize: 60,
    // lineHeight: 76,
    transformOrigin: 'top middle',
    transform: [{ rotate: '-1deg' }],
  },
  subtitle: {
    fontFamily: 'Montserrat',
    fontSize: 18,
    fontWeight: 'bold',
  },
  link: {
    fontFamily: 'Montserrat',
    lineHeight: 30,
    fontSize: 16,
    color: '#0a7ea4',
  },
});
