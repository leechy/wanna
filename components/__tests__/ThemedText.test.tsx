import * as React from 'react';
import { render } from '@testing-library/react-native';

import { ThemedText } from '../ThemedText';

describe('ThemedText', () => {
  it(`renders correctly`, () => {
    const tree = render(<ThemedText>Snapshot test!</ThemedText>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`title renders correctly`, () => {
    const tree = render(
      <ThemedText type="title">Snapshot test!</ThemedText>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`subtitle renders correctly`, () => {
    const tree = render(
      <ThemedText type="subtitle">Snapshot test!</ThemedText>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
