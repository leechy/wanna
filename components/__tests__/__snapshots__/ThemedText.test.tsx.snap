// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedText renders correctly 1`] = `
<Text
  adjustsFontSizeToFit={false}
  numberOfLines={0}
  style={
    [
      {
        "color": "#333333",
      },
      {
        "fontFamily": "Montserrat",
        "fontSize": 17,
        "fontWeight": "500",
        "letterSpacing": -0.2,
        "lineHeight": 24,
        "textAlign": "left",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`ThemedText subtitle renders correctly 1`] = `
<Text
  adjustsFontSizeToFit={false}
  numberOfLines={0}
  style={
    [
      {
        "color": "#333333",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      {
        "fontFamily": "Montserrat",
        "fontSize": 18,
        "fontWeight": "bold",
      },
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`ThemedText title renders correctly 1`] = `
<Text
  adjustsFontSizeToFit={true}
  numberOfLines={1}
  style={
    [
      {
        "color": "#3a1fc1",
      },
      undefined,
      {
        "fontFamily": "GreatVibes-Regular",
        "fontSize": 60,
        "lineHeight": 76,
        "transform": [
          {
            "rotate": "-1deg",
          },
        ],
        "transformOrigin": "top middle",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
